sim/dex-data.ts

```diff
@@ -90,105 +92,82 @@ export class BasicEffect {
 	 * necessarily the current gen being simulated.) Not all effects
 	 * track generation; this will be 0 if not known.
 	 */
-	gen: number;
+	gen: number; // TODO: Generation
 	/**
 	 * Is this item/move/ability/pokemon unreleased? True if there's
 	 * no known way to get access to it without cheating.
 	 */
 	isUnreleased: boolean;
 	/**
-	 * A shortened form of the description of this effect. Not all effects have this
+	 * A shortened form of the description of this effect.
+	 * Not all effects have this.
 	 */
 	shortDesc: string;
-	/**
-	 * The full description for this effect
-	 */
+	/** The full description for this effect. */
 	desc: string;
 	/**
 	 * Is this item/move/ability/pokemon nonstandard? True for effects
 	 * that have no use in standard formats: made-up pokemon (CAP),
 	 * glitches (Missingno etc), and Pokestar pokemon.
 	 */
 	isNonstandard: boolean;
-	/**
-	 * The duration of the effect.
-	 */
+	/** The duration of the effect.  */
 	duration?: number;
-	/**
-	 * Whether or not the effect is ignored by Baton Pass.
-	 */
+	/** Whether or not the effect is ignored by Baton Pass. */
 	noCopy: boolean;
-	/**
-	 * Whether or not the effect affects fainted Pokemon.
-	 */
+	/** Whether or not the effect affects fainted Pokemon. */
 	affectsFainted: boolean;
-	/**
-	 * The status that the effect may cause.
-	 */
-	status?: string;
-	/**
-	 * The weather that the effect may cause.
-	 */
-	weather?: undefined;
-	/**
-	 * HP that the effect may drain.
-	 */
-	drain?: number[];
+	/** The status that the effect may cause. */
+	status?: string; // TODO: Status
```

Comment:
Similarly, `Status` and `Weather` seem like they could be enumerated/typed.

Reply:
No, those definitely shouldn't. PS supports users adding their own statuses and weathers to the data files.

Reply:
Right, I tend to overlook PS's support for non standard.